Doc/library/argparse.rst

```diff
@@ -26,6 +26,61 @@ module also automatically generates help and usage messages and issues errors
 when users give the program invalid arguments.
 
 
+Summary
+-------
+
+Core Functionality
+^^^^^^^^^^^^^^^^^^
+
+.. csv-table::
+    :header: "Name", "Commonly Used Arguments", "Example"
+
+    ":class:`argparse.ArgumentParser`", "`prog`_, `description`_, `formatter_class`_", "``parser = argparse.ArgumentParser(prog='PROG', description='DESC', formatter_class=argparse.RawDescriptionHelpFormatter)``"
+    ":func:`ArgumentParser.add_argument`", "`name or flags`_, `action`_, `default`_, `type`_, `required`_, `help`_", "``parser.add_argument('-v', '--verbose', action='store_true', help='Show various debugging information')``"
+
```

Comment:
To repeat my comment from #1210, maybe this would be nicer to read and to edit as a couple sentences rather than a table?

Reply:
Hi merwok, 

I've changed the table into a couple of sentences as you recommended! Let me know if you have any suggestions for improvement.

Reply:
Well I was thinking aloud and wanted to see what you think :) 

Does it read better?  Should all four tables be tables or paragraphs?

Reply:
Personally, I liked the suggestion of making the table under the Core Functionality a paragraph! However, I think the information in the other tables is more clear and succinct when they are in tables and not paragraphs. 
Let me know what you think! If you prefer paragraphs, I can reformat the other tables into paragraphs.

Reply:
That seems nice!  People landing on this page will see two small paragraph with a direct code example, then the summary tables with direct links to more info on each param.